torch/_higher_order_ops/utils.py

```diff
@@ -197,6 +204,45 @@ def check_alias(out):
     return _detect_input_alias(gm)
 
 
+def _has_potential_branch_output_alias(branch, inputs, pre_dispatch=False):
+    """
+    Dispatch-trace the branch with inputs and check if
+    producing graph has aliasing the branch output. This is
+    bit restrictive as the branch must be traceable.
+    """
+    gm = _create_gm_from_branch(branch, inputs, pre_dispatch)
+    # this can happen when nested cond_op is
+    # functionalized
+    if isinstance(gm, bool):
+        return gm
+
+    def _detect_output_alias(gm):
```

Comment:
We could merge this function with _has_potential_branch_input_alias (change its anme to _has_potential_bracnh_input_outputs_alias, to avoid tracing the subgraph multiple times

Reply:
I merged the functionalities as you suggested and updated all the usages.